packages/gotrue/lib/src/gotrue_mfa_api.dart

```diff
@@ -28,39 +28,57 @@ class GoTrueMFAApi {
 
   /// Starts the enrollment process for a new Multi-Factor Authentication (MFA) factor.
   /// This method creates a new `unverified` factor.
-  /// To verify a factor, present the QR code or secret to the user and ask them to add it to their authenticator app.
   ///
-  /// The user has to enter the code from their authenticator app to verify it.
+  /// For TOTP: To verify a factor, present the QR code or secret to the user and ask them to add it to their authenticator app.
+  /// For Phone: The user will receive an SMS with a verification code.
+  ///
+  /// The user has to enter the code from their authenticator app or SMS to verify it.
   ///
   /// Upon verifying a factor, all other sessions are logged out and the current session's authenticator level is promoted to `aal2`.
   ///
   /// [factorType] : Type of factor being enrolled.
   ///
-  /// [issuer] : Domain which the user is enrolled with.
+  /// [issuer] : Domain which the user is enrolled with (TOTP only).
   ///
   /// [friendlyName] : Human readable name assigned to the factor.
+  ///
+  /// [phone] : Phone number to enroll for Phone factor type.
   Future<AuthMFAEnrollResponse> enroll({
     FactorType factorType = FactorType.totp,
     String? issuer,
     String? friendlyName,
+    String? phone,
   }) async {
```

Comment:
I wonder if it isn't better to deprecate this method and add 2 new ones such as:

```dart
Future<AuthMFAEnrollTOTPResponse> enrollWithTOTP({
    String issuer,
    Stirng? friendlyName,
}) async

Future<AuthMFAEnrollPhoneResponse> enrollWithPhone({
    String phone,
    Stirng? friendlyName,
}) async
```

Notice the method returns different response types, which will have the totp and phone properties non-null respectively.

@dshukertjr what do you think?